### ISSUE TITLE
Enunciate with Resteasy and Guice

### ISSUE THREAD

**Issue by tomasbozzo-everis** [stoicflame/enunciate#603]
Hi,

I'm having trouble trying to generate the documentation for this very simple Resteasy+Guice web app. I've tried different things but none seem to work, I always get this message:

> [WARNING] [ENUNCIATE] No documentation generated: there are no data types, services, or resources to document.

pom.xml
```
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.tomasing</groupId>
    <artifactId>guicetest1</artifactId>
    <packaging>war</packaging>
    <version>0.0.1-SNAPSHOT</version>
    <name>guicetest1 Maven Webapp</name>
    <url>http://maven.apache.org</url>

    <properties>
        <maven.compiler.source>1.7</maven.compiler.source>
        <maven.compiler.target>1.7</maven.compiler.target>
        <enunciate.version>2.8.0</enunciate.version>
    </properties>

    <dependencies>
        <dependency>
            <groupId>com.webcohesion.enunciate</groupId>
            <artifactId>enunciate-core-annotations</artifactId>
            <version>${enunciate.version}</version>
        </dependency>

        <dependency>
            <groupId>com.webcohesion.enunciate</groupId>
            <artifactId>enunciate-rt-util</artifactId>
            <version>${enunciate.version}</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-databind</artifactId>
            <version>2.3.3</version>
        </dependency>
        <dependency>
            <groupId>org.jboss.resteasy</groupId>
            <artifactId>resteasy-guice</artifactId>
            <version>2.3.6.Final</version>
        </dependency>

        <dependency>
            <groupId>aopalliance</groupId>
            <artifactId>aopalliance</artifactId>
            <version>1.0</version>
        </dependency>

        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <version>3.8.1</version>
            <scope>test</scope>
        </dependency>

        <dependency>
            <groupId>javax</groupId>
            <artifactId>javaee-api</artifactId>
            <scope>provided</scope>
            <version>6.0</version>
        </dependency>
    </dependencies>
    <build>
        <finalName>guicetest1</finalName>
        <plugins>
            <plugin>
                <groupId>com.webcohesion.enunciate</groupId>
                <artifactId>enunciate-maven-plugin</artifactId>
                <version>${enunciate.version}</version>
                <executions>
                    <execution>
                        <id>assemble</id>
                        <goals>
                            <goal>assemble</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</project>

```
web.xml
```
<web-app xmlns="http://java.sun.com/xml/ns/javaee" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee 
                        http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd"
         version="3.0">

    <display-name>Archetype Created Web Application</display-name>

    <context-param>
        <param-name>resteasy.guice.modules</param-name>
        <param-value>com.tomasing.guicetest1.modules.HelloModule</param-value>
    </context-param>

    <listener>
        <listener-class>
            org.jboss.resteasy.plugins.guice.GuiceResteasyBootstrapServletContextListener
        </listener-class>
    </listener>

    <servlet>
        <servlet-name>Resteasy</servlet-name>
        <servlet-class>
            org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher
        </servlet-class>
    </servlet>

    <servlet-mapping>
        <servlet-name>Resteasy</servlet-name>
        <url-pattern>/*</url-pattern>
    </servlet-mapping>
</web-app>
```
enunciate.xml

```
<?xml version="1.0"?>
<enunciate xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
           xsi:noNamespaceSchemaLocation="http://enunciate.webcohesion.com/schemas/enunciate-2.8.0.xsd">

    <api-classes>
        <include pattern="com.tomasing.guicetest1.**"/>
    </api-classes>

    <modules>
        <docs docsDir="restApiDocs"/>
    </modules>

</enunciate>
```
HelloController.java
```
package com.tomasing.guicetest1.controllers;

import com.google.inject.Inject;
import com.tomasing.guicetest1.bean.Hello;
import com.tomasing.guicetest1.managers.HelloManager;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;

/**
 * Hello controller
 */
@Path("hello")
public class HelloController {

    private final HelloManager helloManager;

    @Inject
    public HelloController(HelloManager helloManager) {
        this.helloManager = helloManager;
    }

    @GET
    @Path("{name}")
    @Produces("application/json")
    public Hello hello(@PathParam("name") final String name) {
        return new Hello(name);
    }
}
```

After running mvn enunciate:assemble I get the following output

> [INFO] Scanning for projects...
> [INFO] ------------------------------------------------------------------------
> [INFO] Building guicetest1 Maven Webapp
> [INFO]    task-segment: [com.webcohesion.enunciate:enunciate-maven-plugin:2.8.0:assemble]
> [INFO] ------------------------------------------------------------------------
> [INFO] [enunciate:assemble {execution: default-cli}]
> [INFO] [ENUNCIATE] Using enunciate configuration at C:\dev\repository\guicetest1\enunciate.xml
> [INFO] [ENUNCIATE] No JAXB XML data types: C XML client will not be generated.
> [INFO] [ENUNCIATE] No JAXB XML data types: C# XML client will not be generated.
> [INFO] [ENUNCIATE] No Jackson JSON data types: GWT JSON overlays will be generated.
> [INFO] [ENUNCIATE] No Jackson JSON data types: Java JSON client will not be generated.
> [INFO] [ENUNCIATE] No JAXB XML data types: Java XML client will not be generated.
> [INFO] [ENUNCIATE] No Jackson JSON data types: JavaScript client will not be generated.
> [INFO] [ENUNCIATE] No JAXB XML data types: Objective-C XML client will not be generated.
> [INFO] [ENUNCIATE] No Jackson JSON data types: PHP JSON client will not be generated.
> [INFO] [ENUNCIATE] No JAXB XML data types: PHP XML client will not be generated.
> [INFO] [ENUNCIATE] No Jackson JSON data types: Ruby JSON client will not be generated.
> [INFO] [ENUNCIATE] No resource APIs registered: Swagger UI will not be generated.
> [WARNING] [ENUNCIATE] No documentation generated: there are no data types, services, or resources to document.
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD SUCCESSFUL
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 4 seconds
> [INFO] Finished at: Tue Feb 28 14:15:11 CET 2017
> [INFO] Final Memory: 37M/529M
> [INFO] ------------------------------------------------------------------------

The web app is working fine. The URL http://localhost:8080/guicetest1/hello/John returns the Json: {"name":"John"}

Thank you very much for your help.

**Comment by stoicflame** [comment]
What does the output of `mvn -X` say about `com.tomasing.guicetest1.controllers.HelloController`?

**Comment by tomasbozzo-everis** [comment]
Hi,

Thanks for the fast response.

About the output, I get a lot of compiler errors like this one:

```
[DEBUG] [ENUNCIATE] [javac] [ERROR] com.webcohesion.enunciate.Enunciate$URLFileObject[file:/C:/dev/repository/guicetest1/src/main/java/com/tomasing/guicetest1/controllers/HelloController.java]:3:25 /C:/dev/repository/guicetest1/src/main/java/com/tomasing/guicetest1/controllers/HelloController.java:3: error: package com.google.inject does not exist
import com.google.inject.Inject;
                        ^
```

Regards.

**Comment by stoicflame** [comment]
It looks like you need to include the guice annotations on your classpath.

It is curious why the build didn't fail, though.

**Comment by tomasbozzo-everis** [comment]
Not only is the Guice library missing, I get the same error for the **GET**, **Path** and **Produces** annotations. It seems like this issue is related to this one (https://github.com/stoicflame/enunciate/issues/200), but the difference is that my classpath is not empty:

```
[DEBUG] [ENUNCIATE] Compiler classpath: [
  C:\dev\repository\guicetest1\target\classes
]
```

**Comment by stoicflame** [comment]
So does it work when you fix up your classpath?

**Comment by tomasbozzo-everis** [comment]
I'm not sure how to do it :sweat_smile:

**Comment by tomasbozzo-everis** [comment]
Solved it.

It seems that the version of Enunciate I'm using does not support Maven 2, which is necessary for my client's project (shame on them).

I followed the Getting Started guide and only the XML services were being documented, the Persona service was completely ignored because of the same compilation errors I was getting. After building with Maven 3 the Persona service appeared in the documentation.

So I tried building our project's documentation with Maven 3 and it worked fine.

Thank you very much for your help.
